python/cudf/cudf/core/_internals/timezones.py

```diff
@@ -193,15 +193,36 @@ def localize(
     return cast(
         DatetimeTZColumn,
         build_column(
-            data=gmt_data.data,
+            data=gmt_data.base_data,
             dtype=dtype,
-            mask=localized.mask,
+            mask=localized.base_mask,
             size=gmt_data.size,
             offset=gmt_data.offset,
         ),
     )
 
 
+def convert(data: DatetimeTZColumn, zone_name: str) -> DatetimeTZColumn:
+    if not isinstance(data, DatetimeTZColumn):
+        raise TypeError(
+            "Cannot convert from timezone-naive timestamps to "
+            "timezone-aware timestamps. For that, "
+            "use `tz_localize instead."
+        )
+    utc_time = data._utc_time
```

Comment:
if `zone_name` is equal to the current column tz, could short circuit and return a copy?

Reply:
Ah good call! Done